lib/domain/entity/state_management/project_state_manager.dart

```diff
@@ -0,0 +1,21 @@
+import 'package:onix_flutter_bricks/domain/service/strategy/state_manager_strategy.dart';
+import 'package:onix_flutter_bricks/domain/service/strategy/strategies/bloc_strategy.dart';
+import 'package:onix_flutter_bricks/domain/service/strategy/strategies/base_strategy.dart';
+import 'package:onix_flutter_bricks/domain/service/strategy/strategies/provider_strategy.dart';
+
+enum ProjectStateManager {
+  bloc,
+  provider,
+  base;
+
+  StateManagerStrategy get strategy {
```

Comment:
StateManagerStrategy is a generic type. Should it be used as generic if the type is not explicitly specified or used anywhere?